qiskit_experiments/test/fake_backend.py

```diff
@@ -12,38 +12,34 @@
 
 """Fake backend class for tests."""
 import uuid
-from qiskit.providers.backend import BackendV1
-from qiskit.providers.models import QasmBackendConfiguration
+from qiskit.circuit.library import Measure
+from qiskit.providers.backend import BackendV2
+from qiskit.providers.options import Options
+from qiskit.transpiler import Target
 
 from qiskit.result import Result
 
-from qiskit_experiments.framework import Options
 from qiskit_experiments.test.utils import FakeJob
 
 
-class FakeBackend(BackendV1):
+class FakeBackend(BackendV2):
     """
     Fake backend for test purposes only.
     """
 
-    def __init__(self, **config):
-        default_config = {
-            "backend_name": "fake_backend",
-            "backend_version": "0",
-            "n_qubits": int(1e6),
-            "basis_gates": [],
-            "gates": [],
-            "local": True,
-            "simulator": True,
-            "conditional": False,
-            "open_pulse": False,
-            "memory": False,
-            "max_shots": int(1e6),
-            "max_experiments": None,
-            "coupling_map": None,
-        }
-        default_config.update(**config)
-        super().__init__(QasmBackendConfiguration(**default_config))
+    target = None
```

Comment:
Overriding abstract property method in this way seems hacky. I prefer to define a protected member and return it through property method as long as the base class provides such mechanism.

Reply:
I agree that is more proper. I copied this from MinimalBackend, so I updated both classes in this PR now. I think for MinimalBackend I had been thinking I wanted to do `backend.target = Target()` in tests to test out different Target configurations but that wasn't a good reason to deviate from it being a property.